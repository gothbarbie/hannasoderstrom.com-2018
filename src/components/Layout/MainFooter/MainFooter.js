import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

import Logo from '../../Logo'
import SocialMediaSmall from '../../SocialMediaSmall'
import MainNavigation from '../MainNavigation'

import imgTheresMore from '../../../images/theres-more.svg'

const Footer = styled.footer`
  padding: 2rem;
  background: #313131;
  position: relative;

  @media (min-width: 800px) {
    padding: 5% 0;
  }
`

const Linked = styled(Link)`
  text-decoration: none;
  margin-bottom: 3vw;
  align-self: flex-start;

  @media (min-width: 800px) {
    margin-bottom: 0;
  }
`

const Img = styled.img`
  position: absolute;
  right: 14%;
  top: -10%;
  display: none;

  @media (min-width: 640px) {
    display: block;
  }

  @media (min-width: 800px) {
    top: -40%;
  }
`

const FooterWrapper = styled.div`
  max-width: 1140px;
  display: flex;
  flex-direction: ${({ column }) => column && 'column'};
  justify-content: center;
  margin: 0 auto;
  padding: 1rem 4%;
  flex-wrap: wrap;
  position: relative;

  nav {
    margin-top: 5%;

    @media (min-width: 640px) {
      margin-top: 0;
    }
  }

  section {
    @media (min-width: 800px) {
      margin-right: 5%;
    }
  }
`

const LeftArea = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;

  @media (min-width: 640px) {
    align-items: start;
  }

  @media (min-width: 800px) {
    flex-direction: row;
    justify-content: space-between;
  }
`

const MainFooter = () => (
  <Footer>
    <FooterWrapper>
      <LeftArea>
        <Linked to="/">
          <Logo />
        </Linked>
        <SocialMediaSmall />
      </LeftArea>
      <MainNavigation />
      <Img src={imgTheresMore} />
    </FooterWrapper>
  </Footer>
)

export default MainFooter
