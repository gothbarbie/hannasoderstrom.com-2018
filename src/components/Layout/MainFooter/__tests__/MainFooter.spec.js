import React from 'react'
import { shallow } from 'enzyme'

import MainFooter from '../index'

describe('components/MainFooter', () => {
  let component

  beforeEach(() => {
    component = shallow(<MainFooter />)
  })

  it('renders', () => {
    expect(component).toMatchSnapshot()
  })
})
