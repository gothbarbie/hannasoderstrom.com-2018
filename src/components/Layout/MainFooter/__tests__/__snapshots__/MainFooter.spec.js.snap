// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MainFooter renders 1`] = `
<styled.footer>
  <Styled(styled.div)
    maxWidth="1340px"
  >
    <styled.div>
      <Logo
        darkColor={true}
      />
    </styled.div>
    <SocialMedia
      darkColor={true}
    />
  </Styled(styled.div)>
</styled.footer>
`;
